pkg/discovery/dtofactory/node_entity_dto_builder.go

```diff
@@ -221,11 +226,108 @@ func (builder *nodeEntityDTOBuilder) getNodeCommoditiesSold(node *api.Node, clus
+			// If we are missing capacity its unlikely we would have other metrics either
+			continue
+		}
+
+		if resource == "rootfs" {
+			// We iterate the vstorageResources slice in order so the rootfs values
+			// are always preserved in the first pass of this loop.
+			rootfsCapacityBytes = capacityBytes.Avg
+		}
+		// Capacity metric is always a single data point. Use Avg to refer to the single point value
+		commSoldBuilder.Capacity(util.Base2BytesToMegabytes(capacityBytes.Avg))
+
+		usedBytes := float64(0)
+		availableBytes, err := builder.metricValue(metrics.NodeType, entityID,
+			metrics.VStorage, metrics.Available, nil)
+		if err != nil {
+			glog.Warningf("Missing used value for %v : %s for node %s.", resourceType, resource, nodeName)
+		} else {
+			if resource == "rootfs" {
+				rootfsAvailableBytes = availableBytes.Avg
+			}
+			usedBytes = capacityBytes.Avg - availableBytes.Avg
+			commSoldBuilder.Used(util.Base2BytesToMegabytes(usedBytes))
+			commSoldBuilder.Peak(util.Base2BytesToMegabytes(usedBytes))
+		}
+
+		// set commodity key
+		commSoldBuilder.Key(fmt.Sprintf("k8s-node-%s", resource))
+		resourceCommoditySold, err := commSoldBuilder.Create()
+		if err != nil {
+			glog.Warning(err.Error())
+			continue
+		}
+
+		threshold, err := builder.metricValue(entityType, entityID,
+			resourceType, metrics.Threshold, nil)
+		if err != nil {
+			glog.Warningf("Missing threshold value for %v for node %s.", resourceType, entityID)
+		} else {
+
+			if threshold.Avg > 0 && threshold.Avg <= 100 {
+				isAvailableAboveThreshold := availableBytes.Avg > threshold.Avg*capacityBytes.Avg/100
+				isAvailableForPlacement = isAvailableAboveThreshold
+				utilizationThreshold := 100 - threshold.Avg
+				// TODO: The settable method for UtilizationThresholdPct can be added to the sdk instead.
+				resourceCommoditySold.UtilizationThresholdPct = &utilizationThreshold
+			} else {
+				glog.Warningf("Threshold value [%.2f] outside range and will not be set for %v : %s for node %s.",
+					threshold.Avg, resourceType, resource, entityID)
+			}
+		}
+
+		// We currently have no way of knowing the command line configuration of kubelet
+		// to understand if there is a separate imagefs partition configured. We use the workaround
+		// comparing the reported capacity and available bytes, to the last byte, for both
+		// rootfs and imagefs to determine if we are getting the reported values for the same partition.
+		isPartitionSame := resource == "imagefs" && rootfsCapacityBytes == capacityBytes.Avg && rootfsAvailableBytes == availableBytes.Avg
+		if isPartitionSame {
+			// We skip adding imagefs commodity, however we still honor the thresholds set for imagefs
+			// which can be different compared to rootfs, even when the partitions are same.
+			// isAvailableForPlacement is still calculated for both above and would be set to false
+			// if either of the values cross threshold.
+			continue
+		}
```

Comment:
nit: Looks like `isPartitionSame` is not an unusual case, maybe move this section above to around line 287 so we can continue early if needed?

Reply:
Thanks @ading1977 for the suggestion. This is however intentionally here, because we want the thresholds to be evaluated even when its the same partition.